src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -460,11 +471,13 @@ public void extract(
       Object archive,
       Object output,
       String stripPrefix,
+      String strip_prefix,
       Dict<?, ?> renameFiles, // <String, String> expected
       StarlarkThread thread)
       throws RepositoryFunctionException, InterruptedException, EvalException {
     StarlarkPath archivePath = getPath("extract()", archive);
-
+    stripPrefix = (stripPrefix=="''") ? (strip_prefix) : (stripPrefix) ;
```

Comment:
I don't think this works. The Java variable `stripPrefix` will be filled with the Starlark value of the `stripPrefix` parameter, which will almost never be the string `''` (it'll likely be the empty string instead). Could you add a test that exercises the new `strip_prefix` parameter?

Reply:
This image shows that stripPrefix is initialised with "''", does that still imply and empty string ?


<img width="461" alt="Screenshot 2023-01-06 at 8 00 13 AM" src="https://user-images.githubusercontent.com/69615898/210918048-e5c76c17-5c7c-4bb4-b3f4-934afd1fd7b7.png">

Reply:
That's different. The `defaultValue` field is interpreted as Starlark code, so the Java string `"''"` gets turned into the empty string `""`. When you look at the value of the `stripPrefix` variable inside the Java method, it will be the empty string (`""`).

Reply:
In any case, we should really add a test to ensure this doesn't break users.